backend/component/iam/manager.go

```diff
@@ -0,0 +1,43 @@
+package iam
+
+import (
+	_ "embed"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+)
+
+//go:embed acl.yaml
+var aclYaml []byte
+
+type acl struct {
+	Allowlist []Binding `yaml:"allowlist"`
+}
+
+type Binding struct {
+	Role        string   `yaml:"role"`
+	Permissions []string `yaml:"permissions"`
+}
+
+type Manager struct {
+	Allowlist map[string][]Permission
+}
+
+func NewManager() (*Manager, error) {
+	predefinedACL := new(acl)
+	if err := yaml.Unmarshal(aclYaml, predefinedACL); err != nil {
+		return nil, errors.Wrapf(err, "failed to unmarshal predefined acl")
+	}
+
+	allowlist := make(map[string][]Permission)
+	for _, binding := range predefinedACL.Allowlist {
+		allowlist[binding.Role] = make([]Permission, len(binding.Permissions))
```

Comment:
Avoid using size hint unless performance is a concern.
https://google.github.io/styleguide/go/best-practices#vardeclsize

Reply:
>  It is acceptable to preallocate when the final size is known (e.g. when converting between a map and a slice) but this is not a readability requirement, and may not be worth the clutter in small cases.

I failed to find your quote.

Reply:
Even though “this is not a readability requirement”, we should maintain a consistent and one usage in the codebase, preallocate the slice without length hint.